src/ipbb/cmds/repo.py

```diff
@@ -285,6 +285,14 @@ def git(env, repo, branch, dest):
         echo('Checking out branch/tag ' + style(branch, fg='blue'))
         sh.git('checkout', branch, '-q', _out=sys.stdout, _cwd=lRepoLocalPath)
 
+    elif revision is not None:
+        echo('Checking out revision ' + style(revision, fg='blue'))
+        try:
+            sh.git('checkout', revision, '-q', _out=sys.stdout, _cwd=lRepoLocalPath)
+        except Exception as err:
+            secho("Failed to check out requested revision. Staying on master.", fg='red')
```

Comment:
Minor detail, `master` is not guaranteed to be present (the default branch is defined by the repository).

Reply:
From playing around on the command line a bit, it appears git 'does the right thing': in case one tries to checkout a non-existent branch or revision the local repo appears to remain unchanged. So no explicit revert is necessary.